React/apis-get.md

```diff
@@ -1,26 +1,81 @@
-1. Next, we'll install the axios package by running `npm install axios --save`.
-1. Verify! Open `package.json` and ensure we see axios in the "dependencies" section.
+2. Next check out the `ada-pets-api` branch with `git checkout ada-pets-api`.  This will load a version of Ada Pets which is already working, but using the imported json file.
+3. Next, we'll install the axios package by running `npm install axios --save`.
+4. Verify! Open `package.json` and ensure we see axios in the "dependencies" section.
+
+### Review The Code
+
+Now review the code and see how this application is structured it may work a little differently than your implementation.  Specifically the state data is all in the `App` component.
+
+### Running the Server
+
+We have the json-server package pre-installed and set up in our `package.json` file.  Open up `package.json` and look for the following section:
+
+```json
+"scripts": {
+    "start": "react-scripts start",
+    "server": "json-server --watch src/data/pets.json --port 2999",
+    "build": "react-scripts build",
+    "test": "react-scripts test",
+    "eject": "react-scripts eject"
+  },
+```
+
+Check out the `server` line.  This adds a command to start the json-server package with the port 2999.  You can open the api in the browser with [http://localhost:2999/pets](http://localhost:2999/pets).  You can do all the RESTful routes for a pets resource, listing showing, creating etc.  All these operations will update the pets.json file.  So `pets.json` will serve as an imitation API for us to learn with.
+
+**Exercise**
+
+With your neighbor, and using Postman, access the API to list and show several pets.  
+
+Answer: 
+
+- What is the URL to list all pets?
+- What is the URL to list a single pet?
+- What happens when you try to access a pet that does not exist?
+
+### Replacing the Static Pets Data
 
 Now we are set up to load axios library code within our React application using an `import` statement.
 
+```javascript
+// App.js
+import axios from 'axios';
+```
+
 Let's update our pets app to remove the hard-coded data to start:
 1. Remove the array from the `App` component
-1. Remove the `prop` passed from the `App` component to the `PetCollection` component
-1. Update the `PetCollection` constructor to no longer use the `props` passed in to the set the original state
-1. Set the initial state of `pets` to an empty array
+2. Remove the `prop` passed from the `App` component to the `PetList` component
+3. Set the initial state of `petList` to an empty array.
+
+```javascript
+// App.js
+class App extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      petList: [],
+      currentPet: undefined,
+      originalPets: [],
+    };
+  }
+```
 
 ### Recall axios Structure
-Recall the structure of the `axios` request from our earlier time using it with jQuery:
+
+Recall the structure of the `axios` request from our earlier time using it with Node:
+
 ```javascript
 axios.get('url')
```

Comment:
I'd like to put `setResult` and `setError` inside of the `.then` and `.catch` to draw the parallels.